dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/condition/config/model/ConditionRuleParser.java

```diff
@@ -40,14 +44,24 @@
  */
 public class ConditionRuleParser {
 
-    public static ConditionRouterRule parse(String rawRule) {
+    public static AbstractRouterRule parse(String rawRule) {
+        AbstractRouterRule rule;
         Yaml yaml = new Yaml(new SafeConstructor(new LoaderOptions()));
         Map<String, Object> map = yaml.load(rawRule);
-        ConditionRouterRule rule = ConditionRouterRule.parseFromMap(map);
-        rule.setRawRule(rawRule);
-        if (CollectionUtils.isEmpty(rule.getConditions())) {
-            rule.setValid(false);
+        String confVersion = (String) map.get(CONFIG_VERSION_KEY);
+
+        if (confVersion != null && confVersion.startsWith(RULE_VERSION_V31)) {
+            rule = MultiDestConditionRouterRule.parseFromMap(map);
+            if (CollectionUtils.isEmpty(((MultiDestConditionRouterRule) rule).getConditions())) {
+                rule.setValid(false);
+            }
+        } else {
+            rule = ConditionRouterRule.parseFromMap(map);
+            if (CollectionUtils.isEmpty(((ConditionRouterRule) rule).getConditions())) {
+                rule.setValid(false);
+            }
```

Comment:
If rule version bigger than 3.1, should skip and print warning log.

Reply:
I have added that logic in the new version